test/library/draft/Vector/Vector.chpl

```diff
@@ -0,0 +1,1263 @@
+      const size = lst.size;
+
+      on this {
+        _enter();
+        result = _insertGenericKnownSize(idx, lst, size);
+        _leave();
+      }
+
+      return result;
+    }
+
+    /*
+      Insert a vector of elements `vec` into this vector at index `idx`, shifting
+      all elements at and following the index `vec.size` positions to the
+      right.
+
+      If the insertion is successful, this method returns `true`. If the given
+      index is out of bounds, this method does nothing and returns `false`.
+
+      .. warning::
+
+        Inserting elements into this vector may invalidate existing references
+        to the elements contained in this vector.
+
+      :arg idx: The index into this vector at which to insert.
+      :type idx: `int`
+
+      :arg lst: A vector of elements to insert.
+      :type lst: `vector(eltType)`
+
+      :return: `true` if `vec` was inserted, `false` otherwise.
+      :rtype: `bool`
+    */
+    proc ref insert(idx: int, vec: vector(eltType)): bool lifetime this < vec {
+      
+      var result = false;
+      
+      // Prevent deadlock if we are trying to insert this into itself.
+      const size = vec.size;
+
+      on this {
+        _enter();
+        result = _insertGenericKnownSize(idx, vec, size);
+        _leave();
+      }
+
+      return result;
+    }
+
+    /*
+      Remove the first `count` elements from this vector with values equal to
+      `x`, shifting all elements following the removed item left.
+
+      If the count of elements to remove is less than or equal to zero, then
+      all elements from this vector equal to the value of `x` will be removed.
+
+      .. warning::
+
+        Removing elements from this vector may invalidate existing references
+        to the elements contained in this vector.
+
+      :arg x: The value of the element to remove.
+      :type x: `eltType`
+
+      :arg count: The number of elements to remove.
+      :type count: `int`
+
+      :return: The number of elements removed.
+      :rtype: `int`
+    */
+    proc ref remove(x: eltType, count:int=1): int {
+      var result = 0;
+
+      on this {
+        var src_i, dst_i: int;
+        var removed = 0;
+        src_i = 0;
+
+        _enter();
+
+        // Run through the leading non-x prefix
+        while dst_i < _size {
+          if _data[dst_i] == x {
+            removed += 1;
+            break;
+          }
+          dst_i += 1;
+        }
+
+        // Once we've found an x, everything after has to be _move()d.
+        // Run src_i ahead, moving everything that isn't x back to dst_i.
+        src_i = dst_i + 1;
+        while src_i < _size {
+          if (count == 0 || removed < count) && _data[src_i] == x {
+            src_i += 1;
+            removed += 1;
+            continue;
+          }
+          _data[dst_i] = _data[src_i];
+          src_i += 1;
+          dst_i += 1;
+        }
+
+        if (removed) {
+          _size = _size - removed;
+        }
```

Comment:
Shouldn't we free the allocated memory as well?

Reply:
Or delete the reference to these.

Reply:
https://github.com/chapel-lang/chapel/pull/16048/commits/3315421c5416589be2d9351f69c31274a338c91b